src/backend/executor/execMain.c

```diff
@@ -3056,7 +3056,7 @@ ExecPostprocessPlan(EState *estate)
  * ----------------------------------------------------------------
  */
 void
-ExecEndPlan(PlanState *planstate, EState *estate)
+ExecEndPlan(PlanState *planstate, EState *estate, CmdType operation)
```

Comment:
how about to change it to
```suggestion
ExecEndPlan(PlanState *planstate)
```
because PlanState contains both EState and CmdType?

Reply:
`Planstate` has pointer to `estate`, but does not have field `operation`. And for execution `ExecEndPlan`, pointer to `estate` is get from `queryDesc`. Do you sure, that these are the same pointers at all time?

Reply:
> `Planstate` has pointer to `estate`, but does not have field `operation`. And for execution `ExecEndPlan`, pointer to `estate` is get from `queryDesc`. Do you sure, that these are the same pointers at all time?

I corrected suggestion, look more carefully

Reply:
done.